priv/perf/apps/load_test/lib/service/sync.ex

```diff
@@ -12,36 +12,45 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-defmodule LoadTest.Ethereum.Sync do
+defmodule LoadTest.Service.Sync do
   @moduledoc """
   Provides a function for repeating a function call until a given criteria is met.
   """
-  @sleep_interval_ms 100
+
+  require Logger
+
+  import LoadTest.Service.Sleeper
 
   @doc """
   Repeats f until f returns {:ok, ...}, :ok OR exception is raised (see :erlang.exit, :erlang.error) OR timeout
   after `timeout` milliseconds specified
 
   Simple throws and :badmatch are treated as signals to repeat
   """
-  def repeat_until_success(f, timeout) do
-    fn -> do_repeat_until_success(f) end
+  def repeat_until_success(f, timeout, message) do
```

Comment:
🤔 I know this is old code logic, but do we not want a circuit breaker after certain times of retries??

Reply:
Most of the usages of this function are related for waiting for some condition/even in the chain. 

For cases where an external service is not available, I think we can implement some kind of circuit breaker that will fail immediately.